src/main/java/org/cryptomator/common/mountpoint/CustomMountPointChooser.java

```diff
@@ -60,21 +70,47 @@ public boolean prepare(Volume caller, Path mountPoint) throws InvalidMountPointE
 				throw new InvalidMountPointException(new IllegalStateException("Not implemented"));
 			}
 		}
-		LOG.debug("Successfully checked custom mount point: {}", mountPoint);
-		return false;
 	}
 
-	private void prepareParentNoMountPoint(Path mountPoint) throws InvalidMountPointException {
-		//This the case on Windows when using FUSE
-		//See https://github.com/billziss-gh/winfsp/issues/320
-		Path parent = mountPoint.getParent();
-		if (!Files.isDirectory(parent)) {
-			throw new InvalidMountPointException(new NotDirectoryException(parent.toString()));
-		}
-		//We must use #notExists() here because notExists =/= !exists (see docs)
-		if (!Files.notExists(mountPoint, LinkOption.NOFOLLOW_LINKS)) {
-			//File exists OR can't be determined
-			throw new InvalidMountPointException(new FileAlreadyExistsException(mountPoint.toString()));
+	//This the case on Windows when using FUSE
+	//See https://github.com/billziss-gh/winfsp/issues/320
+	void prepareParentNoMountPoint(Path mountPoint) throws InvalidMountPointException {
+		Path hideaway = getHideaway(mountPoint);
+		var mpExists = Files.exists(mountPoint);
+		var hideExists = Files.exists(hideaway);
```

Comment:
Instead of multiple `Files.exists` and `Files.isDirectory` invocations, you might want to issue a single `mountPointAttr = Files.readAttributes(mountPoint, BasicFileAttributes.class, NOFOLLOW_LINKS)` for either path, setting the result to `null` in case of a `NoSuchFileException`.

This avoids redundant I/O.

Reply:
I looked at the code again for such refactor it, but i decided against it.

We need at least 2 calls to the fs (mountpoint and hideaway). And in every brach we make at most one additional call for probing if it is a directory. Also, directly reading the attributes would require additional try-catch-blocks or a change in the logic (`hideaway` should either be the only one existing or not exist at all, the secdon throws for `readAttributes`, the second depends on mp, making it hard to write it easy readable), which is not worth for the scenario imho.